core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4578,6 +4582,14 @@ private void validateVersionPointer(QualifiedObjectName tableName, QueryPeriod q
                 if (pointer == null) {
                     throw semanticException(INVALID_ARGUMENTS, queryPeriod, "Pointer value cannot be NULL");
                 }
+                long pointerTimestampMillis = unpackMillisUtc((long) coerce(type, pointer, createTimestampWithTimeZoneType(3)));
+                if (pointerTimestampMillis >= System.currentTimeMillis()) {
```

Comment:
Compare with session start time.
This will make `AS OF now()` queries fail deterministically. (Otherwise they will work sometimes, but not always)

Reply:
Addressed.

However, i decided to do the math with maximum precision. The `AS OF now()` has millisecond precision and thus may be rounded to a milli that's sub-milli before the session start, thus may or may not be rejected.
The `AS OF now(9)` (or higher precision) will deterministically be rejected, since session start time is measured with nano precision (`Instant`).